Zend/zend_language_parser.y

```diff
@@ -633,6 +635,36 @@ case_separator:
 ;
 
 
+match:
+		T_MATCH '(' expr ')' '{' match_arm_list '}'
+			{ $$ = zend_ast_create(ZEND_AST_MATCH, $3, $6); };
+;
+
+match_arm_list:
+		%empty { $$ = zend_ast_create_list(0, ZEND_AST_MATCH_ARM_LIST); }
+	|	non_empty_match_arm_list possible_comma { $$ = $1; }
+;
+
+non_empty_match_arm_list:
+		match_arm { $$ = zend_ast_create_list(1, ZEND_AST_MATCH_ARM_LIST, $1); }
+	|	non_empty_match_arm_list ',' match_arm { $$ = zend_ast_list_add($1, $3); }
+;
+
+match_arm:
+		match_arm_cond_list T_DOUBLE_ARROW match_arm_body { $$ = zend_ast_create(ZEND_AST_MATCH_ARM, $1, $3); }
+	|	T_DEFAULT T_DOUBLE_ARROW match_arm_body { $$ = zend_ast_create(ZEND_AST_MATCH_ARM, NULL, $3); }
+;
+
+match_arm_cond_list:
+		expr { $$ = zend_ast_create_list(1, ZEND_AST_EXPR_LIST, $1); }
+	|	match_arm_cond_list ',' expr { $$ = zend_ast_list_add($1, $3); }
+;
```

Comment:
What about
```php
match ($x) {
    FOOBAR,
    BARFOO,
        => true,
    default,
        => false
}
```
?

This is pretty ugly, but I'm not really seeing what the canonical formatting for multiline match condition lists is supposed to be.

Reply:
Done. I thought this was already supported but apparently not.